src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -415,7 +422,12 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
           language: textTracks[i].language,
           index: i
         };
-        if (settings.language || settings.label) {
+        if (settings.kind === 'subtitles') {
+          parsedTracks.push(new PKTextTrack(settings));
+        }
+        if (settings.kind === 'captions' && this._config.enableCEA708Captions) {
+          settings.label = settings.label || captionsTextTrackLabels.pop();
```

Comment:
can we do better then pop-ing the array? seems a bit hack-ish.

Reply:
what bad with `pop` (actually should be `shift`)? it'a a legitimate api.